input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go

```diff
@@ -131,6 +131,8 @@ func SetStructValues(in interface{}, values *Values, opts ...SetStructValuesOpti
 			fieldVal = reflect.ValueOf(values.Str)
 		case reflect.Int, reflect.Int64:
 			fieldVal = reflect.ValueOf(values.Int).Convert(f.Type())
+		case reflect.Float64:
+			fieldVal = reflect.ValueOf(values.Float).Convert(f.Type())
```

Comment:
Why is this necessary?

Reply:
It is because Event.SuccessCount.Sum is a float64 and in [SetStructValues in populator.go](https://github.com/elastic/apm-data/blob/main/input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go#L126), the default value is first initialized by field type before checking if the field should be excluded (which it should)
```
--- FAIL: TestDecodeNestedError (0.00s)
    --- FAIL: TestDecodeNestedError/decode (0.00s)
panic: unhandled type float64 for key Event.SuccessCount.Sum [recovered]
        panic: unhandled type float64 for key Event.SuccessCount.Sum

goroutine 21 [running]:
testing.tRunner.func1.2({0xb60c00, 0xc0001d4c30})
        /home/carson/sdk/go1.19.3/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
        /home/carson/sdk/go1.19.3/src/testing/testing.go:1399 +0x39f
panic({0xb60c00, 0xc0001d4c30})
        /home/carson/sdk/go1.19.3/src/runtime/panic.go:884 +0x212
github.com/elastic/apm-data/input/elasticapm/internal/modeldecoder/modeldecodertest.SetStructValues.func1({0xb57840?, 0xc0001ce850?, 0xc0002a4d00?}, {0xc000169fb0, 0x16})
        /home/carson/projects/apm-data/input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go:235 +0x185d
github.com/elastic/apm-data/input/elasticapm/internal/modeldecoder/modeldecodertest.iterateStruct({0xbd0680?, 0xc0001ce848?, 0xc0002a4fa0?}, {0xc000169f68, 0x12}, 0xc0002a53c0)
        /home/carson/projects/apm-data/input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go:374 +0x314
github.com/elastic/apm-data/input/elasticapm/internal/modeldecoder/modeldecodertest.iterateStruct({0xc219c0?, 0xc0001ce830?, 0xc0002a5240?}, {0xc000249f28, 0x5}, 0xc0002a53c0)
        /home/carson/projects/apm-data/input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go:386 +0x60f
github.com/elastic/apm-data/input/elasticapm/internal/modeldecoder/modeldecodertest.iterateStruct({0xc6c140?, 0xc0001ce800?, 0x4?}, {0x0, 0x0}, 0xc0002a53c0)
        /home/carson/projects/apm-data/input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go:386 +0x60f
github.com/elastic/apm-data/input/elasticapm/internal/modeldecoder/modeldecodertest.IterateStruct({0xbaaf80?, 0xc0001ce800?}, 0xc000161448?)
        /home/carson/projects/apm-data/input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go:347 +0xc5
github.com/elastic/apm-data/input/elasticapm/internal/modeldecoder/modeldecodertest.SetStructValues({0xbaaf80?, 0xc0001ce800?}, 0x0?, {0xc0002a5428?, 0x0?, 0xc7c5e0?})
        /home/carson/projects/apm-data/input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go:127 +0x5f
github.com/elastic/apm-data/input/elasticapm/internal/modeldecoder/v2.initializedInputMetadata(_)
        /home/carson/projects/apm-data/input/elasticapm/internal/modeldecoder/v2/metadata_test.go:188 +0xcc
github.com/elastic/apm-data/input/elasticapm/internal/modeldecoder/v2.TestDecodeNestedError.func1(0xc00024d6c0?)
        /home/carson/projects/apm-data/input/elasticapm/internal/modeldecoder/v2/error_test.go:49 +0x85
testing.tRunner(0xc00024d6c0, 0xcccb18)
        /home/carson/sdk/go1.19.3/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
        /home/carson/sdk/go1.19.3/src/testing/testing.go:1493 +0x35f
FAIL    github.com/elastic/apm-data/input/elasticapm/internal/modeldecoder/v2   0.007s
```